### ISSUE TITLE
Bug: `-d` tag adding space to dir name

### ISSUE THREAD

**Issue by halleycarleton** [jekyll/jekyll#3666]
When running `jekyll build -d _dir-name` it's adding a space to the beginning of the directory name. So instead of writing the build to "_dir-name" it's writing it to " _dir-name".

In order to get it to build to the correct directory I have to run this instead: `jekyll build -d_dir-name`

*Ubuntu 14.04*
*Jekyll 2.5.3*
*Ruby 2.2.0*

**Comment by envygeeks** [comment]
Could this be a bug in mercenary instead of Jekyll considering it's mercenary that should handle opts?

**Comment by halleycarleton** [comment]
I don't really know. I'm just reporting it here because I experienced it using jekyll.

**Comment by envygeeks** [comment]
Sorry, I meant to @parkr that so he could answer since he has experience with both!

**Comment by parkr** [comment]
We just [use `OptionParser#parse!`](https://github.com/jekyll/mercenary/blob/master/lib/mercenary/program.rb#L31) so maybe it's just a fact of how that works? I agree it's unintuitive/annoying/unclear.

**Comment by envygeeks** [comment]
I've never had that happen in `OptParse`.

**Comment by halleycarleton** [comment]
If this syntax is how it is supposed to work I think the documents should be updated to reflect that.

Here's a screenshot of the issue:
![Bug Screenshot]
(http://i.imgur.com/DOppvbK.png?1)

You can see that the flag is entered as `-d_test` and the destination directory is `_test`.

**Comment by parkr** [comment]
It's a bug, you should be able to either specify or leave out the directory. Funny that this has never happened to me. @envygeeks can you repro?

[Here's where we fetch option values.](https://github.com/jekyll/mercenary/blob/master/lib/mercenary/command.rb#L180-L182)

**Comment by walutisme** [comment]
![2015-04-23_04-51-10](https://cloud.githubusercontent.com/assets/10618799/7285104/cd2d1b64-e974-11e4-8fe3-de30cb2c8de3.png)

Doesnt happen to me

14.04
Jekyll 2.5.3
Ruby 2.2.2

**Comment by parkr** [comment]
What shells are you two using? Bash? ZSH?

**Comment by halleycarleton** [comment]
Bash here

**Comment by envygeeks** [comment]
I was just unable to reproduce it on my Bash from here on GNU/Linux.

**Comment by envygeeks** [comment]
Is there source available? I would certainly love to try to see if I can reproduce it from your source.

**Comment by halleycarleton** [comment]
Can you specify what you mean by source?

**Comment by envygeeks** [comment]
Jekyll needs source to build, if you can supply your site source so I can test it then please do, if you cannot then fair enough we'll probably have to close because it's hard to reproduce it, but if you cannot provide the source but can somehow find a way to reproduce the problem we can fix the problem.

**Comment by halleycarleton** [comment]
Well I can't even replicate it on the source I was originally having the problem with. It seems to have gone away on it's own. I can't think of anything that has changed since I first noticed the issue, no software updates, changes to the source, or even a reboot.

The only thing I can think of is that I ran jekyll on a fresh test project to try to replicate the issue, but I don't know if that's even related.

Oh well, thanks for taking the time to help.
